Balint-list2.py

```diff
@@ -0,0 +1,66 @@
+#!/usr/bin/python -tt
+# Copyright 2010 Google Inc.
+# Licensed under the Apache License, Version 2.0
+# http://www.apache.org/licenses/LICENSE-2.0
+
+# Google's Python Class
+# http://code.google.com/edu/languages/google-python-class/
+
+# Additional basic list exercises
+
+# D. Given a list of numbers, return a list where
+# all adjacent == elements have been reduced to a single element,
+# so [1, 2, 2, 3] returns [1, 2, 3]. You may create a new list or
+# modify the passed in list.
+def remove_adjacent(nums):
+    return sorted(set(nums))
+
+
+# E. Given two lists sorted in increasing order, create and return a merged
+# list of all the elements in sorted order. You may modify the passed in lists.
+# Ideally, the solution should work in "linear" time, making a single
+# pass of both lists.
+def linear_merge(list1, list2):
```

Comment:
ok, you seem to have missed the point. The initial list are already sorted. Try to use that information to achieve linear time sorting.